iroha_p2p/src/peer.rs

```diff
@@ -183,10 +200,85 @@ where
         let read_half = self.read.as_mut().unwrap();
         let public_key = read_server_hello(read_half).await?;
         self.derive_shared_key(&public_key)?;
-        self.state = State::Ready;
+        self.state = State::SendKey;
+        Ok(())
+    }
+
+    /// Sends our app public key
+    async fn send_our_public_key(&mut self) -> Result<(), Error> {
+        debug!("Sending our public key...");
+        #[allow(clippy::unwrap_used)]
+        let write_half = self.write.as_mut().unwrap();
+        write_half.as_ref().writable().await?;
+
+        // We take our public key from this field and will replace it with theirs when we read it
+        #[allow(clippy::unwrap_used)]
+        let pub_key = self.id.public_key.take().unwrap();
```

Comment:
Better to use expect and to explain inside of it why take always returns Some()

Reply:
Hm, you gave me another thought.
Previously, we had the time when we had no public key in this field, but now we have keys there every time. I'll get rid of option part.